### DISCUSSION TITLE
Post F´ v3.0.0 Roadmap

### DISCUSSION THREAD

**Discussion by ThibFrgsGmz** [nasa/fprime#1271] [Q&A] [answered]
Hello,

Is it possible to have a clear view of F´'s roadmap for the coming years please?

According to the "GitHub Project" tab, the user does not have a clear overview (no epic/release/sprint scheduled).
The "Post v3.0.0 Development" project suggests to me that there has been a very significant spike in team activity but there is nothing really planned for the future in terms of features.

Should we expect a slowdown in the development of the project consisting of fixing the bugs in the current version for a certain period of time (1 year?)?

In the GitHub discussions, we heard about the addition of ``Data Product`` functionality in the medium term. What about it?

Following @bocchino's presentation on ``FPP`` at yesterday's ``FSW Workshop``, I asked a question on this topic, which unfortunately went unanswered due to time constraints:

> Should we expect in the long run to (macroscopically?) define the system topology through a ``WebGUI``?
> That is, without having to write FPP files, and get closer to what is done in ESTEC's ``TASTE`` framework.

Is it possible to put in a ``ROADMAP.md`` the currently considered roadmap?
We would see the features seriously considered to be added with a time and release target.

PS: do you plan to use the new [Projects / Issues](https://github.com/features/issues) features of GitHub, for now in beta I think?

**Comment by ThibFrgsGmz** [comment]
I don't know if a file containing the roadmap is the best choice, here is an example with the [Flutter's roadmap](https://github.com/olexale/flutter_roadmap).
Alternatively, there could be a GitHub dashboard, similar to the [Slack's roadmap](https://trello.com/b/ZnTQyumQ/slack-platform-roadmap).

**Comment by Joshua-Anderson** [comment]
I second this! As a open source project, I think having a shared roadmap makes it easier for members of the community to know what contributors are working on and collaborate in ways the support these milestones (or avoid investing development in an area that's likely to change in upcoming releases).

**Comment by LeStarch** [comment, marked as answer]
I am not opposed to publishing a roadmap.  However, at this time the formal F´ roadmap is pretty short:

1. Release `v3.1.0` to address build system performance
2. Ongoing F´ v3.x.x maintenance

We can add bigger features and mid/long term goals (e.g. Data Products) once they've formalized enough that we have an understanding of what the feature development schedule looks like.
